content-src/asrouter/templates/Trailhead/Trailhead.jsx

```diff
@@ -131,9 +132,33 @@ export class _Trailhead extends React.PureComponent {
     return str.value;
   }
 
+  onCardAction(action) {
+    let actionUpdates = {};
+
+    if (action.type === "OPEN_URL") {
+      let url = new URL(action.data.args);
+      let params = new URLSearchParams(url.search.slice(1));
+      params.append("utm_source", "activity-stream");
+      params.append("utm_campaign", "firstrun");
+      params.append("utm_medium", "referral");
+      params.append("utm_term", this.props.message.utm_term);
+
+      if (action.addFlowParams) {
+        params.append("flow_id", this.state.flowId);
+        params.append("flow_begin_time", this.state.flowBeginTime);
+      }
+
+      url = `${url.href.split("?")[0]}?${params.toString()}`;
```

Comment:
In general, we shouldn't split or build URLs manually. As noted above, if we use `url.searchParams`, we shouldn't need this line (and the `{args: url}` below should just work as part of `URL.toString()`